cmd/mdatagen/internal/templates/config.go.tmpl

```diff
@@ -12,7 +12,8 @@ import (
 
 // MetricConfig provides common config for a particular metric.
 type MetricConfig struct {
-	Enabled bool `mapstructure:"enabled"`
+	Enabled bool   `mapstructure:"enabled"`
+	Name    string `mapstructure:"name"`
```

Comment:
Why are we adding this to the end-user config while it doesn't have any effect if user changes it? AFAIU goal of https://github.com/open-telemetry/opentelemetry-collector/issues/12459 is to have the metric name generated and available for a component developer, not end user.

Reply:
fixed in https://github.com/open-telemetry/opentelemetry-collector/pull/12531/commits/030ce9ec3cc9ba9d8e295444215562723c856e0f

Reply:
fixed in https://github.com/open-telemetry/opentelemetry-collector/pull/12531/commits/030ce9ec3cc9ba9d8e295444215562723c856e0f

Reply:
fixed in https://github.com/open-telemetry/opentelemetry-collector/pull/12531/commits/030ce9ec3cc9ba9d8e295444215562723c856e0f